### PR TITLE
Upgrading monitoring module to use Kubernetes 1.22 runtime

### PR DESCRIPTION
**Pull request by nandajavarma**

### CODE REVIEW

File: katalog/prometheus-operated/kube-prometheus-rules.yml [hunk 1]
```diff
@@ -7,63 +7,68 @@ apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
   labels:
+    app.kubernetes.io/component: exporter
+    app.kubernetes.io/name: kube-prometheus
+    app.kubernetes.io/part-of: kube-prometheus
     prometheus: k8s
     role: alert-rules
   name: kube-prometheus-rules
+  namespace: monitoring
```
Comment (angelbarrera92) [inline katalog/prometheus-operated/kube-prometheus-rules.yml:16]: Avoid adding the `namespace` name. It is injected via the `kustomization.yaml` file.

File: katalog/prometheus-operated/kube-prometheus-rules.yml [hunk 2]
```diff
@@ -7,63 +7,68 @@ apiVersion: monitoring.coreos.com/v1
...
-        - expr: sum(rate(node_cpu_seconds_total{mode!="idle",mode!="iowait",mode!="steal"}[5m]))
-          record: cluster:node_cpu:sum_rate5m
-        - expr: cluster:node_cpu_seconds_total:rate5m / count(sum(node_cpu_seconds_total) BY (instance, cpu))
-          record: cluster:node_cpu:ratio
-    - name: kube-prometheus-general.rules
-      rules:
-        - expr: count without(instance, pod, node) (up == 1)
-          record: count:up1
-        - expr: count without(instance, pod, node) (up == 0)
-          record: count:up0
```
Comment (angelbarrera92) [inline katalog/prometheus-operated/kube-prometheus-rules.yml:69]: Is it possible to maintain the alignment so it is easier to compare/review the changes?

File: katalog/prometheus-operated/kube-prometheus-rules.yml [hunk 3]
```diff
@@ -7,63 +7,68 @@ apiVersion: monitoring.coreos.com/v1
...
+    - alert: TargetDown
+      annotations:
+        description: '{{ printf "%.4g" $value }}% of the {{ $labels.job }}/{{ $labels.service }} targets in {{ $labels.namespace }} namespace are down.'
+        runbook_url: https://runbooks.prometheus-operator.dev/runbooks/general/targetdown
+        summary: One or more targets are unreachable.
+      expr: 100 * (count(up == 0) BY (job, namespace, service) / count(up) BY (job, namespace, service)) > 10
+      for: 10m
+      labels:
+        severity: warning
+    - alert: Watchdog
```
Comment (angelbarrera92) [inline katalog/prometheus-operated/kube-prometheus-rules.yml:30]: `Watchdog` is one of the "special" rules that we modify from upstream. If you take a look carefully, we have it renamed to `DeadMansSwitch` and must continue as `DeadMansSwitch`.

File: katalog/tests/tests.sh [hunk 4]
```diff
@@ -6,15 +6,44 @@
...
+}
+
+@test "cert-manager is Running" {
+  info
+  test() {
+    kubectl get pods -l app=cert-manager -o json -n cert-manager | jq '.items[].status.containerStatuses[].ready' | uniq | grep -q true
+  }
+  loop_it test 60 10
+  status=${loop_it_result:?}
+  [ "$status" -eq 0 ]
```
Comment (angelbarrera92) [inline katalog/tests/tests.sh:46]: Nice! I like it. Thanks!

File: katalog/prometheus-operated/deploy.yml [hunk 5]
```diff
@@ -31,10 +31,7 @@ spec:
       cpu: 2000m
       memory: 2Gi
   retention: 30d
-  ruleSelector:
-    matchLabels:
-      prometheus: k8s
-      role: alert-rules
+  ruleSelector: {}
```
Comment (angelbarrera92) [inline katalog/prometheus-operated/deploy.yml:34]: Why we are removing this?

**Review by angelbarrera92 — APPROVED** [review]
LGTM!

File: katalog/prometheus-operated/deploy.yml (hunk 5 above)
Comment (nandajavarma) [reply katalog/prometheus-operated/deploy.yml:34]: @angelbarrera92 I removed it to be consistent with the upstream. This [PR](https://github.com/prometheus-operator/kube-prometheus/pull/1315) removes it.

**Pull request merged**